pkg/app/app_diff_test.go

```diff
@@ -0,0 +1,325 @@
+
+	// t.Run("include-needs fail on disabled direct need", func(t *testing.T) {
+	// 	check(t, testcase{
+	// 		fields: fields{
+	// 			skipNeeds:    false,
+	// 			includeNeeds: true,
+	// 		},
+	// 		selectors: []string{"name=test2"},
+	// 		error:     `in ./helmfile.yaml: release "default//test2" depends on "default/kube-system/disabled" which does not match the selectors. Please add a selector like "--selector name=disabled", or indicate whether to skip (--skip-needs) or include (--include-needs) these dependencies`,
+	// 	})
+	// })
+
+	// t.Run("include-needs fail on disabled transitive need", func(t *testing.T) {
+	// 	check(t, testcase{
+	// 		fields: fields{
+	// 			skipNeeds:    false,
+	// 			includeNeeds: true,
+	// 		},
+	// 		selectors: []string{"name=test3"},
+	// 		error:     `in ./helmfile.yaml: release "default//test2" depends on "default/kube-system/disabled" which does not match the selectors. Please add a selector like "--selector name=disabled", or indicate whether to skip (--skip-needs) or include (--include-needs) these dependencies`,
+	// 	})
+	// })
+
+	// t.Run("include-transitive-needs fail on disabled transitive need", func(t *testing.T) {
+	// 	check(t, testcase{
+	// 		fields: fields{
+	// 			skipNeeds:              false,
+	// 			includeNeeds:           false,
+	// 			includeTransitiveNeeds: true,
+	// 		},
+	// 		selectors: []string{"name=test3"},
+	// 		error:     `in ./helmfile.yaml: release "default//test2" depends on "default/kube-system/disabled" which does not match the selectors. Please add a selector like "--selector name=disabled", or indicate whether to skip (--skip-needs) or include (--include-needs) these dependencies`,
+	// 	})
+	// })
+
+	// t.Run("include-needs with include-transitive-needs fail on disabled direct need", func(t *testing.T) {
+	// 	check(t, testcase{
+	// 		fields: fields{
+	// 			skipNeeds:              false,
+	// 			includeNeeds:           true,
+	// 			includeTransitiveNeeds: true,
+	// 		},
+	// 		selectors: []string{"name=test2"},
+	// 		error:     `in ./helmfile.yaml: release "default//test2" depends on "default/kube-system/disabled" which does not match the selectors. Please add a selector like "--selector name=disabled", or indicate whether to skip (--skip-needs) or include (--include-needs) these dependencies`,
+	// 	})
+	// })
+
+	// t.Run("include-needs with include-transitive-needs fail on disabled transitive need", func(t *testing.T) {
+	// 	check(t, testcase{
+	// 		fields: fields{
+	// 			skipNeeds:              false,
+	// 			includeNeeds:           true,
+	// 			includeTransitiveNeeds: true,
+	// 		},
+	// 		selectors: []string{"name=test3"},
+	// 		error:     `in ./helmfile.yaml: release "default//test2" depends on "default/kube-system/disabled" which does not match the selectors. Please add a selector like "--selector name=disabled", or indicate whether to skip (--skip-needs) or include (--include-needs) these dependencies`,
+	// 	})
+	// })
```

Comment:
@mumoshu why comment those codes? Thanks very much.

Reply:
@yxxhero Good catch! It was an artifact of my TODO that needs to be addressed before merging this. I've fixed that in 1735095